test/tests_for_sdkmr/EE2Runjob_test.py

```diff
@@ -0,0 +1,268 @@
+            },
+        },
+        concierge_params=None,
+    )
+
+    # updated job data save
+    mongo.get_job.assert_called_once_with("603051cfaf2e3401b0500982")
+
+    # update to queued state
+    got_job = sdkmr.save_job.call_args_list[1][0][0]
+    expected_job = Job()
+    expected_job.id = ObjectId("603051cfaf2e3401b0500982")
+    expected_job.status = "queued"
+    # no way to test this really without code refactoring
+    expected_job.queued = got_job.queued
+
+    expected_job.scheduler_type = "condor"
+    expected_job.scheduler_id = "cluster42"
+    assert_jobs_equal(got_job, expected_job)
+
+    kafka.send_kafka_message.assert_called_with(  # update to queued state
+        KafkaQueueChange(
+            job_id="603051cfaf2e3401b0500982",
+            new_status="queued",
+            previous_status="created",
+            scheduler_id="cluster42",
+        )
+    )
+    slack.run_job_message.assert_called_once_with(
+        "603051cfaf2e3401b0500982", "cluster42", "someuser"
+    )
+
+
+def assert_jobs_equal(got_job: Job, expected_job: Job):
+    """
+    Checks that the two jobs are equivalent, except that the 'updated' fields are checked that
+    they're within 1 second of each other.
+    """
+    # I could not get assert got_job == expected_job to ever work, or even
+    # JobRequirements1 == JobRequirements2 to work when their fields were identical, so we
+    # do this disgusting hack instead. Note it will need to be updated any time a job field is
+    # added.
+    if not hasattr(got_job, "id"):
+        assert not hasattr(expected_job, "id")
+    else:
+        assert got_job.id == expected_job.id
+    assert got_job.user == expected_job.user
+    assert got_job.authstrat == expected_job.authstrat
+    assert got_job.wsid == expected_job.wsid
+    assert got_job.status == expected_job.status
+
+    # The Job class fills the updated field with the output of time.time on instantiation
+    # so we can't do a straight equality
+    assert abs(got_job.updated - expected_job.updated) < 1
+
+    assert got_job.queued == expected_job.queued
+    assert got_job.estimating == expected_job.estimating
+    assert got_job.running == expected_job.running
+    assert got_job.finished == expected_job.finished
+    assert got_job.errormsg == expected_job.errormsg
+    assert got_job.msg == expected_job.msg
+    assert got_job.error == expected_job.error
+    assert got_job.terminated_code == expected_job.terminated_code
+    assert got_job.error_code == expected_job.error_code
+    assert got_job.scheduler_type == expected_job.scheduler_type
+    assert got_job.scheduler_id == expected_job.scheduler_id
+    assert got_job.scheduler_estimator_id == expected_job.scheduler_estimator_id
+
+    if not got_job.job_input:
+        assert expected_job.job_input is None
+    else:
+        assert got_job.job_input.wsid == expected_job.job_input.wsid
+        assert got_job.job_input.method == expected_job.job_input.method
+        assert (
```

Comment:
Did black add these brackets in? If not, have you ensured that `assert` is doing what you think it is?
e.g. see the 'Caveat #2' section of https://dbader.org/blog/python-assert-tutorial

What a great idea it was to make a programming language hypersensitive to whitespace. 😖

Reply:
> Did black add these brackets in? 

Yes. >:

Reply:
None of them could be interpreted as tuples so I think it's ok in this case, but that's a good warning. I'll add a comment to that effect.

Reply:
I deliberately made one fail just to be safe

Reply:
> far more sensible JavaScript.

I'm rolling